File: ClientApp/src/app/views/reporting/models/vacancy-report-dto.ts
export class VacancyReportDto {
    totalPost : number = 0;
    totalInService : number = 0;
    totalVacant : number = 0;
    vacancyDetailsDto: VacancyDetailsDto[] = [];
}
export class VacancyDetailsDto {
    departmentName : string = "";
    sectionName : string = "";
    designationName : string = "";
    totalPost : number = 0;
    totalInService : number = 0;
    totalVacantPost : number = 0;

    departmentId : number = 0;
    sectionSequence : number = 0;
    designationPosition : number = 0;
}

File: Hrm.Application/DTOs/Reporting/VacancyReportDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hrm.Application.DTOs.Reporting
{
    public class VacancyReportDto
    {
        public int? TotalPost { get; set; }
        public int? TotalInService { get; set; }
        public int? TotalVacant { get; set; }
        public List<VacancyDetailsDto> VacancyDetailsDto { get; set; }
    }
    public class VacancyDetailsDto
    {
        public string? DepartmentName { get; set; }
        public string? SectionName { get; set; }
        public string? DesignationName { get; set; }
        public int? TotalPost { get; set; }
        public int? TotalInService { get; set; }
        public int? TotalVacantPost { get; set; }

        public int? DepartmentId { get; set; }
        public int? SectionSequence { get; set; }
        public int? DesignationPosition { get; set; }
    }
}
